packages/mui-base/src/Slider/Slider.tsx

```diff
@@ -156,14 +156,22 @@ const Slider = React.forwardRef(function Slider<RootComponentType extends React.
     className: classes.rail,
   });
 
+  let adjustedTrackOffset = trackOffset;
```

Comment:
I would suggest moving this logic to the hook itself. This way it can be re-used in all components that depend on the hook (both Joy UI and Mateiral UI)

Reply:
Wouldn't this be suitable for a new issue? I would prefer if this PR gets merged, and that I can work on new issue which consists of moving the logic to the hook instead, since this issue only targets the mui-base code

Reply:
> since this issue only targets the mui-base code

The hooks are part of the Base UI code and the hooks are used in all components, so we would benefit fixing the issue in all projects. I don't see why we should delay this.

Reply:
@michaldudak What do you think? I still think that the provided solution is appropriate for the issue, and a new issue, which can be assigned to me would be suitable for moving the logic to the hook.

Reply:
@michaldudak ping :)

Reply:
I agree with @mnajdova. Most of the logic should be placed inside hooks, with components being just a thin layer on top of them providing structure.

Reply:
@michaldudak @mnajdova Can you take a look at the PR now? I think it passes all the requirements, since the failing test seems okay either way. Or do you have any way of solving it?